src/main/java/com/conveyal/datatools/manager/jobs/MergeFeedsJob.java

```diff
@@ -233,6 +243,58 @@ public void jobFinished() {
             tablesToMerge.addAll(Arrays.asList(GtfsPlusTable.tables));
         }
         int numberOfTables = tablesToMerge.size();
+        // Before initiating the merge process, run some pre-processing to check for id conflicts for certain tables
+        if (mergeType.equals(SERVICE_PERIOD)) {
+            mergeStrategy = getMergeStrategy(feedsToMerge);
+            if (mergeStrategy == CHECK_STOP_TIMES) {
+                Feed futureFeed = new Feed(DataManager.GTFS_DATA_SOURCE, feedsToMerge.get(0).version.namespace);
+                Feed pastFeed = new Feed(DataManager.GTFS_DATA_SOURCE, feedsToMerge.get(1).version.namespace);
+                for (String tripId : intersectingTripIds) {
+                    // Fetch all ordered stop_times for each common trip_id, hash them, and compare the two sets for the
+                    // future and current feed. If the stop_times are an exact match, include one instance of the trip
+                    // (ignoring the other identical one). If they do not match, modify the current trip_id and include.
+                    List<StopTime> futureStopTimes = Lists.newArrayList(futureFeed.stopTimes.getOrdered(tripId));
+                    List<StopTime> pastStopTimes = Lists.newArrayList(pastFeed.stopTimes.getOrdered(tripId));
+                    String futureServiceId = futureFeed.trips.get(tripId).service_id;
+                    String pastServiceId = pastFeed.trips.get(tripId).service_id;
+                    // FIXME: what if service_ids do not match! Perhaps the right approach would be to just return
+                    //  FAIL_DUE_TO_MATCHING_TRIP_IDS in that case. It might be too complicated otherwise.
+                    if (!stopTimesMatch(futureStopTimes, pastStopTimes)) {
+                        // If stop_times or services do not match, the trip will be cloned. Also, track the service_id
+                        // (it will need to be cloned and renamed for both current/past feeds).
+                        tripIdsToModifyForCurrentFeed.add(tripId);
+                        serviceIdsToCloneAndRename.add(futureServiceId);
+                    } else {
+                        // If the trip's stop_times are an exact match, we can safely include just the
+                        // future trip and exclude the current/past one. Also, track the service_id (it will need to be
+                        // extended to the full time range).
+                        tripIdsToSkipForCurrentFeed.add(tripId);
+                        serviceIdsToExtend.add(futureServiceId);
+                    }
+                }
+                for (String tripId : tripsOnlyInCurrentFeed) {
+                    String serviceId = pastFeed.trips.get(tripId).service_id;
+                    // If a trip in the current feed contains a service_id that has been modified above, it needs to be
```

Comment:
Suggestion: '**that service_id** needs to be cloned ** and renamed to differentiate it from the same service_id in the future feed. (The trip in question will be linked to the cloned service_id.)**'